### PR TITLE
Email campaign context key

### PR DESCRIPTION
**Pull request by ilappe**
## Status
- [ ] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: https://github.com/demisto/etc/issues/35684

## Description
FindEmailCampaign - let the user choose which fields to see in the context output

## Minimum version of Demisto
- [x] 5.0.0
- [ ] 5.5.0
- [ ] 6.0.0

## Does it break backward compatibility?
   - [ ] Yes
   - [x] No

## Must have
- [x] Tests
- [x] Documentation

### CODE REVIEW

File: Packs/Campaign/Scripts/FindEmailCampaign/FindEmailCampaign_test.py [hunk 1]
```diff
@@ -162,3 +165,41 @@ def test_return_campaign_details_entry_list_dumped_recipients_cc(mocker):
...
+ADDITIONAL_CONTET_KEYS_PARAMETRIZE = [
+    (['name', 'emailfrom', 'emailto', 'severity', 'status', 'created'], []),
+    (['name_', 'emailfrom', 'emailto', 'severity1', 'status', 'created'], ['name_', 'severity1']),
+    ([], [])
+]
+
+
+@pytest.mark.parametrize(
+    'expected_in_context, not_expected_in_context', ADDITIONAL_CONTET_KEYS_PARAMETRIZE)
+def test_context_populated_with_required_keys_happy_path(mocker, expected_in_context, not_expected_in_context):
```
Comment (teizenman) [inline Packs/Campaign/Scripts/FindEmailCampaign/FindEmailCampaign_test.py:179]: Add the Given-When-Then annotation of docstring

File: Packs/Campaign/Scripts/FindEmailCampaign/FindEmailCampaign_test.py [hunk 2]
```diff
@@ -162,3 +165,41 @@ def test_return_campaign_details_entry_list_dumped_recipients_cc(mocker):
...
+
+    # assert that valid expected keys are in the context
+    # and invalid keys aren't in the context
+    assert context['EmailCampaign.incidents'][0]
+    for original_incident, context_incident in zip(incidents_list, context['EmailCampaign.incidents']):
+        for k in expected_in_context:
+            if k in not_expected_in_context:
+                assert context_incident.get(k) is None and original_incident.get(k) is None
+            else:
+                assert context_incident[k] is not None and original_incident[k] == context_incident[k]
```
Comment (teizenman) [inline Packs/Campaign/Scripts/FindEmailCampaign/FindEmailCampaign_test.py:205]: I don't get these assertions, improve this code to be more readable.

**Review by teizenman — CHANGES_REQUESTED** [review]

File: Packs/Campaign/Scripts/FindEmailCampaign/FindEmailCampaign.py [hunk 3]
```diff
@@ -342,8 +344,7 @@ def create_email_summary_hr(incidents_df, fields_to_display):
     hr_email_summary += '\n ##### ' + email_summary
     context = add_context_key(
         create_context_for_campaign_details(
-            campaign_found=True,
-            incidents_df=incidents_df,
+            campaign_found=True, incidents_df=incidents_df,
             additional_context_fields=fields_to_display))
```
Comment (teizenman) [inline Packs/Campaign/Scripts/FindEmailCampaign/FindEmailCampaign.py]: It should look like:
```
    context = add_context_key(
        create_context_for_campaign_details(
            campaign_found=True,
            incidents_df=incidents_df,
            additional_context_fields=fields_to_display,
        )
    )
```

File: Packs/Campaign/Scripts/FindEmailCampaign/FindEmailCampaign_test.py [hunk 4]
```diff
@@ -188,18 +185,62 @@ def test_context_populated_with_required_keys_happy_path(mocker, expected_in_con
...
-    assert context['EmailCampaign.incidents'][0]
-    for original_incident, context_incident in zip(incidents_list, context['EmailCampaign.incidents']):
-        for k in expected_in_context:
-            if k in not_expected_in_context:
-                assert context_incident.get(k) is None and original_incident.get(k) is None
-            else:
-                assert context_incident[k] is not None and original_incident[k] == context_incident[k]
+    for context_incident in context['EmailCampaign.incidents']:
+        for field in invalid_fields:
+            assert field not in context_incident, 'the field "{}" should not be stored in context'.format(field)
```
Comment (teizenman) [inline Packs/Campaign/Scripts/FindEmailCampaign/FindEmailCampaign_test.py:246]: You can use the f-string here instead:
https://realpython.com/python-f-strings/

**Review by teizenman — CHANGES_REQUESTED** [review]

File: Packs/Campaign/Scripts/FindEmailCampaign/FindEmailCampaign.py [hunk 5]
```diff
@@ -70,6 +70,13 @@
 }
 
 
+INVALID_KEY_WARNING = 'Warning: the fields {fields} was not found in the phishing incidents. Please make sure that '\
+                      'you\'ve specified the machine-name of the fields. The machine name can be found in the '\
+                      'settings of the incident field you are trying to search '
```
Comment (teizenman) [inline Packs/Campaign/Scripts/FindEmailCampaign/FindEmailCampaign.py:75]:
```suggestion
                      'settings of the incident field you are trying to search.'
```

**Review by teizenman — APPROVED** [review]
Change the warning message and you're good to go

**Review by teizenman — APPROVED** [review]

**Comment by ShirleyDenkberg** [comment]
@teizenman @lperryp Doc review completed.

**Pull request merged**